exporter/main.go

```diff
@@ -1,203 +1,58 @@
-					&core.EventsSelector{
-						WorkerPhases: []core.WorkerPhase{phase},
-					},
-					&meta.ListOptions{},
-				)
-				if err != nil {
-					log.Fatal(err)
-				}
-
-				allWorkersByPhase.With(
-					prometheus.Labels{"workerPhase": string(phase)},
-				).Set(float64(len(eventsList.Items) +
-					int(eventsList.RemainingItemCount)))
-
-				// brigade_all_running_workers_duration
-
-				var jobsList []core.Job
-				for _, worker := range eventsList.Items {
-					if phase == core.WorkerPhaseRunning {
-						allRunningWorkersDuration.With(
-							prometheus.Labels{"worker": worker.ID},
-						).Set(time.Since(*worker.Worker.Status.Started).Seconds())
-						// brigade_pending_jobs_total
-						for _, job := range worker.Worker.Jobs {
-							if job.Status.Phase == core.JobPhasePending {
-								jobsList = append(jobsList, job)
-							}
-						}
-					} else {
-						allRunningWorkersDuration.Delete(
-							prometheus.Labels{"worker": worker.ID},
-						)
-					}
-				}
-
-				// brigade_pending_jobs_total
-				totalPendingJobs.Set(float64(len(jobsList)))
-			}
-
-			// brigade_users_total
-			userList, err := client.Authn().Users().List(
-				context.Background(),
-				&authn.UsersSelector{},
-				&meta.ListOptions{},
-			)
-			if err != nil {
-				log.Fatal(err)
-			}
-
-			totalUsers.Set(float64(len(userList.Items) +
-				int(userList.RemainingItemCount)))
-
-			// brigade_service_accounts_total
-			saList, err := client.Authn().ServiceAccounts().List(
-				context.Background(),
-				&authn.ServiceAccountsSelector{},
-				&meta.ListOptions{},
-			)
-			if err != nil {
-				log.Fatal(err)
-			}
-
-			totalServiceAccounts.Set(float64(len(saList.Items) +
-				int(saList.RemainingItemCount)))
-
-			// brigade_projects_total
-			projectList, err := client.Core().Projects().List(
-				context.Background(),
-				&core.ProjectsSelector{},
-				&meta.ListOptions{},
-			)
-			if err != nil {
-				log.Fatal(err)
-			}
+func main() {
+	log.Printf(
+		"Starting Brigade Metrics Exporter -- version %s -- commit %s",
+		version.Version(),
+		version.Commit(),
+	)
 
-			totalProjects.Set(float64(len(projectList.Items) +
-				int(projectList.RemainingItemCount)))
+	ctx := signals.Context()
 
-			time.Sleep(time.Duration(scrapeInterval) * time.Second)
+	{
+		address, token, opts, err := apiClientConfig()
+		if err != nil {
+			log.Fatal(err)
 		}
-	}()
-}
-
-func initializeClient() (sdk.APIClient, error) {
-	// The address of the Brigade 2 API server
-	// beginning with http:// or https//
-	apiAddress, err := os.GetRequiredEnvVar("API_ADDRESS")
-	if err != nil {
-		return nil, err
-	}
-
-	// An API token obtained using the Brigade 2 CLI
-	apiToken, err := os.GetRequiredEnvVar("API_TOKEN")
-	if err != nil {
-		return nil, err
+		scrapeInterval, err := scrapeDuration()
+		if err != nil {
+			log.Fatal(err)
+		}
+		go recordMetricsLoop(
+			ctx,
+			sdk.NewAPIClient(address, token, &opts),
+			time.Duration(scrapeInterval), // TODO: Fix this
```

Comment:
Should this TODO be removed? I'm not sure what needs to be fixed about it

Reply:
Yes! Nice catch.

Reply:
Fixed